skbio/stats/composition.py

```diff
@@ -608,10 +611,329 @@ def centralize(mat):
+    To summarize the results from the W-statistic, let's take a look and the
+    results from the hypothesis test:
+
+    >>> results['reject']
+    b1    False
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    Name: reject, dtype: bool
+
+    From this we can conclude that only b2 was significantly
+    different between the treatment and the control.
+
+    """
+    if len(table) != len(grouping):
+        raise ValueError('The number of samples in `table` (%d) '
+                         'must be equal to the number of samples in '
+                         '`grouping` (%d).' % (len(table),
+                                               len(grouping)))
+    if not 0 < alpha < 1:
+        raise ValueError('alpha=%f is not within 0 and 1' % alpha)
+
+    if not 0 < tau < 1:
+        raise ValueError('tau=%f is not within 0 and 1' % alpha)
+
+    if not 0 < theta < 1:
+        raise ValueError('theta=%f is not within 0 and 1' % alpha)
+
+    if multiple_comparisons_correction is not None:
+        if multiple_comparisons_correction != 'holm-bonferroni':
+            raise ValueError('%s is not an option for '
+                             '`multiple_comparisons_correction`'
+                             % multiple_comparisons_correction)
+
+    if significance_test is None:
+        significance_test = scipy.stats.f_oneway
+
+    mat = _check_composition(table, ignore_zeros=False)
+    cats = pd.Series(grouping)
+
+    mat.sort_index(inplace=True)
+    cats = cats.sort_index()
+    labs = mat.columns
+
+    n_samp, n_feat = mat.shape
+
+    _logratio_mat = _log_compare(mat.values, cats.values, significance_test)
+    logratio_mat = _logratio_mat + _logratio_mat.T
+
+    # Multiple comparisons
+    if multiple_comparisons_correction == 'holm-bonferroni':
+        logratio_mat = np.apply_along_axis(_holm_bonferroni,
+                                           1, logratio_mat)
+    np.fill_diagonal(logratio_mat, 1)
+    W = (logratio_mat < alpha).sum(axis=1)
+    c_start = W.max() / n_feat
+    if c_start < theta:
+        reject = np.zeros_like(W, dtype=bool)
+    else:
+        # Select appropriate cutoff
+        cutoff = c_start - np.linspace(0.05, 0.25, 5)
+        prop_cut = np.array([(W > n_feat*cut).mean() for cut in cutoff])
+        dels = np.abs(prop_cut - np.roll(prop_cut, -1))
+        dels[-1] = 0
+        if (dels[1] < tau) and (dels[2] < tau) and (dels[3] < tau):
+            nu = cutoff[1]
+        elif (dels[1] >= tau) and (dels[2] < tau) and (dels[3] < tau):
+            nu = cutoff[2]
+        elif (dels[2] >= tau) and (dels[3] < tau) and (dels[4] < tau):
+            nu = cutoff[3]
+        else:
+            nu = cutoff[4]
+        reject = (W >= nu*n_feat)
+    return pd.DataFrame({'W': pd.Series(W, index=labs),
+                         'reject': pd.Series(reject, index=labs)})
+
+
+def _check_composition(x, ignore_zeros=True):
```

Comment:
This function's doing more than it needs to now because `table` is guaranteed to be a DataFrame and `ignore_zeros` is always passed `False`. I think it can be inlined with `ancom`:

```python
table = pd.DataFrame(closure(table), index=table.index, columns=table.columns)
if np.any(table == 0):
    raise ValueError(...)
```